packages/fetch-and-compile/test/fetch.test.ts

```diff
@@ -0,0 +1,118 @@
+import assert from "assert";
+import "mocha";
+import Config from "@truffle/config";
+import { fetchAndCompile } from "../lib/index";
+import axios from "axios";
+import sinon from "sinon";
+const fixture: any = require("./fixture.js");
+
+function stubAxiosGetMethod(url: string, address: string, data: object) {
+  sinon
+    .stub(axios, "get")
+    .withArgs(url, {
+      params: {
+        module: "contract",
+        action: "getsourcecode",
+        address: address,
+        apikey: ""
+      },
+      responseType: "json",
+      maxRedirects: 50
+    })
+    .returns(Promise.resolve({ data: data }));
+}
+
+afterEach(function () {
+  //TS can't detect that is a sinon stub so we have to use ts-ignore
+  //@ts-ignore
+  //restoring stub
+  axios.get.restore();
+});
+
+describe("fetchAndCompile", function () {
+  it("verifies contract from mainnet", async function () {
+    const config = Config.default().merge({
+      networks: {
+        mainnet: {
+          network_id: 1
+        }
+      },
+      network: "mainnet"
+    });
+    const address = "0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D";
+    stubAxiosGetMethod(
+      "https://api.etherscan.io/api",
+      address,
+      fixture.mainnetData
+    );
+    //TS can't detect that is a sinon stub so we have to use ts-ignore
+    //@ts-ignore
+    axios.get.callThrough();
+    const result = await fetchAndCompile(address, config);
+    const contractNameFromResult = result.sourceInfo.contractName;
```

Comment:
Minor suggestion here: Everything comes from `result`, maybethis would be better called `contractNameFromSourceInfo`?  (Same comment for the other two instances, not going to repeat it.)